### PR TITLE
cluster-api 0.3.14 (new formula)

### PR DESCRIPTION
**Pull request by ludusrusso**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is related to kubernetes-sigs/cluster-api/issues/2389

cc @fabriziopandini

### CODE REVIEW

File: Formula/cluster-api.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  desc "Home for the Cluster Management API work, a subproject of sig-cluster-lifecycle"
+  homepage "https://cluster-api.sigs.k8s.io"
+  url "https://github.com/kubernetes-sigs/cluster-api/archive/refs/tags/v0.3.14.tar.gz"
+  sha256 "fc7c7eb07bce70505721c8bc82ff4b8f111172cdfef94964b7ceac82e45faace"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", bin/"clusterctl"
```
Comment (carlocab) [inline Formula/cluster-api.rb:11]:
```suggestion
    system "go", "build", *std_go_args, "-o", bin/"clusterctl"
```

File: Formula/cluster-api.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    help_output = shell_output("#{bin}/clusterctl --help 2>&1", 2)
+    assert_match "Usage of #{bin}/clusterctl", help_output
```
Comment (carlocab) [inline Formula/cluster-api.rb:16]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @ludusrusso.

File: Formula/cluster-api.rb (hunk 1 above)
Comment (dawidd6) [inline Formula/cluster-api.rb:11]:
```suggestion
    system "go", "build", *std_go_args, "-o", bin/"clusterctl"
```

File: Formula/cluster-api.rb (hunk 2 above)
Comment (ludusrusso) [reply Formula/cluster-api.rb:16]: Yes sure! Thanks!! 🙏

Comment (ludusrusso) [reply Formula/cluster-api.rb:16]: Any suggestion on this? This tool is similar to [kubernetes-cli](https://github.com/Homebrew/homebrew-core/blob/master/Formula/kubernetes-cli.rb) and, in this formular, their are checking for help and version.

I can check for an error, but this cli needs a kubernetes cluster to work.

Comment (carlocab) [reply Formula/cluster-api.rb:16]: We have a lot of formulae (usually old ones) with bad tests. We try to fix the ones we can, but there are a lot of them.

Checking for an expected error from running without a kubernetes cluster is fine.

Comment (ludusrusso) [reply Formula/cluster-api.rb:16]: Ok thanks for help! I'll work on it!

Comment (ludusrusso) [reply Formula/cluster-api.rb]: What do you think?

File: Formula/cluster-api.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "-o", bin/"clusterctl"
+  end
+
+  test do
+    help_output = shell_output("KUBECONFIG=/homebrew.config  #{bin}/clusterctl 2>&1", 1)
+    assert_match "invalid configuration: no configuration has been provided", help_output
```
Comment (carlocab) [inline Formula/cluster-api.rb:16]:
```suggestion
    output = shell_output("KUBECONFIG=/homebrew.config  #{bin}/clusterctl 2>&1", 1)
    assert_match "invalid configuration: no configuration has been provided", output
```
Looks good; let's just not call it `help_output` I think that's a bit confusing.

Comment (ludusrusso) [reply Formula/cluster-api.rb]: You are totally right 😂

**Review by dawidd6 — APPROVED** [review]

**Comment by carlocab** [comment]
Thanks, @ludusrusso.

**Comment by ludusrusso** [comment]
Thanks to you @carlocab!
Which is the best way to automatically trigger a new PR on this formula when they release a new version of cluster-api?

**Comment by SMillerDev** [comment]
https://github.com/marketplace/actions/homebrew-bump-formula

**Pull request closed without merging**